lisa/tools/uptime.py

```diff
@@ -26,3 +29,36 @@ def since_time(self, no_error_log: bool = True) -> datetime:
                 f"when run {self.command} -s."
             )
         return parser().parse(command_result.stdout)
+
+    @classmethod
+    def _windows_tool(cls) -> Optional[Type[Tool]]:
+        return WindowsUptime
+
+
+class WindowsUptime(Uptime):
+
+    # 3/8/2022 10:47:19 PM
+    DATETIME_STRING_REGEX = r"\d{1,2}\/\d{1,2}\/\d{4} \d{1,2}:\d{2}:\d{2} \w{2}"
+
+    @property
+    def command(self) -> str:
+        return ""
+
+    def _check_exists(self) -> bool:
+        return True
+
+    def since_time(self, no_error_log: bool = True, timeout: int = 600) -> datetime:
+        powershell = self.node.tools[PowerShell]
+
+        # get the last boot time
+        result = powershell.run_cmdlet(
+            "Get-CimInstance -ClassName win32_operatingsystem | select lastbootuptime",
+            force_run=True,
+            timeout=timeout,
+        )
+
+        # parse the last boot time from regex
+        match = re.search(self.DATETIME_STRING_REGEX, result)
```

Comment:
The datetime format may be different, so the check may return nothing. Try to output the value only, and call `parse().parse` directly.

Reply:
Updated regex to capture entire string without matching a date format explicitly